exec/java-exec/src/main/java/org/apache/drill/exec/store/RecordDataType.java

```diff
@@ -49,15 +55,25 @@
   public final RelDataType getRowType(RelDataTypeFactory factory) {
     final List<SqlTypeName> types = getFieldSqlTypeNames();
     final List<String> names = getFieldNames();
+    final List<Boolean> nullables = getFieldNullability();
     final List<RelDataType> fields = Lists.newArrayList();
-    for (final SqlTypeName typeName : types) {
+    Iterator<SqlTypeName> typesIter = types.listIterator();
+    Iterator<Boolean> nullabilityIter = nullables.listIterator();
```

Comment:
We now four correlated lists. This is often difficult to reason about and maintain.

Is it possible to have a single list of, say, `FieldDefn` objects, each of which has a name, type, nullable and `RelDataType`?

Reply:
I wanted to avoid changing existing implementation, so I added an extra list and iterated in parallel.

Reply:
This is, in fact, the gist of the comment. While it is understandable to want to leave he existing code as-is, adding more lists to the existing parallel lists is not an improvement: it has caused a semi-wrong initial implementation to become overly cumbersome.

If the implementation is local to this class or module, then leaving the code cleaner after your change than when you found it is generally a good thing.